NDB.Covid19/NDB.Covid19/WebServices/AnonymousTokenService.cs

```diff
@@ -0,0 +1,163 @@
+﻿using AnonymousTokens.Client.Protocol;
+using AnonymousTokens.Core.Services;
+using NDB.Covid19.Configuration;
+using NDB.Covid19.OAuth2;
+using Newtonsoft.Json;
+using Org.BouncyCastle.Asn1.X9;
+using Org.BouncyCastle.Crypto.EC;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Math;
+using Org.BouncyCastle.Math.EC;
+using Org.BouncyCastle.Utilities.Encoders;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NDB.Covid19.WebServices
+{
+    public class AnonymousTokenService
+    {
+        private Initiator _initiator = new Initiator();
+        private X9ECParameters _ecParameters;
+
+        public AnonymousTokenService(X9ECParameters ecParameters)
+        {
+            _ecParameters = ecParameters;
+        }
+
+        public async Task<string> GetAnonymousTokenAsync()
+        {
+            var tokenState = GenerateTokenRequest();
+            var tokenResponse = await RequestTokenGeneration(tokenState);
+            var publicKey = await GetPublicKeyAsync(tokenResponse);
+            var token = RandomizeToken(tokenState, tokenResponse, publicKey);
+            return EncodeToken(tokenState, token, tokenResponse.Kid);
+        }
+
+        public AnonymousTokenState GenerateTokenRequest()
+        {
+            var init = _initiator.Initiate(_ecParameters.Curve);
+            return new AnonymousTokenState
+            {
+                t = init.t,
+                r = init.r,
+                P = init.P
+            };
+        }
+
+        public ECPoint RandomizeToken(AnonymousTokenState state, GenerateTokenResponseModel tokenResponse, ECPublicKeyParameters K)
+        {
+            var Q = _ecParameters.Curve.DecodePoint(Hex.Decode(tokenResponse.QAsHex));
```

Comment:
any reason why some variables are in upper case, and some are in lower case?

Reply:
Point on elliptic curves are upper case and other information, like scalars and byte arrays etc., are lower case. This is standard notation in cryptography.

Reply:
The anonymous token service class should be migrated into https://github.com/HenrikWM/anonymous-tokens, where these variable names are used to be unambigious with the crypto code.

Reply:
Can we do that?